samples/client/petstore/python_disallowAdditionalPropertiesIfNotPresent/petstore_api/model_utils.py

```diff
@@ -186,6 +187,30 @@ def __getattr__(self, attr):
         """get the value of an attribute using dot notation: `instance.attr`"""
         return self.__getitem__(attr)
 
+    def __copy__(self):
+        cls = self.__class__
+        if self.get("_spec_property_naming", False):
+            return cls._new_from_openapi_data(**self.__dict__)
+        else:
+            return new_cls.__new__(cls, **self.__dict__)
+
+    def __deepcopy__(self, memo):
+        cls = self.__class__
+        kwargs = {}
+        if self.discriminator and self.discriminator.keys():
+            key = list(cls.discriminator.keys())[0]
+            kwargs[key] = self.get(key)
```

Comment:
Why are we grabbing the discriminator?
Won't it already be in the new instance because it is defined in the classes?

Reply:
I think that now new_from_openapi_data is used, it does the correct thing with the discriminator so this is no longer needed as you suggest. I will remove this.